### PR TITLE
deeper integration

### PR DESCRIPTION
**Pull request by pepyakin**

### CODE REVIEW

**Comment by pepyakin** [comment]
* **#277** <a href="https://app.graphite.dev/github/pr/thrumdev/nomt/277?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @pepyakin and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: nomt/src/bitbox/mod.rs [hunk 1]
```diff
@@ -46,36 +47,39 @@ pub struct Shared {
...
+        let sync_seqn = sync_seqn as u64;
 
-        if !db_path.is_file() {
-            std::fs::create_dir_all(path).unwrap();
-            store::create(db_path.clone(), 64_000).unwrap();
-        }
+        let wal_path = path.join("wal");
 
-        let (store, meta_page, meta_map) = match store::Store::open(db_path) {
+        dbg!();
```
Comment (rphmeier) [inline nomt/src/bitbox/mod.rs:62]: some strays in this file

File: nomt/src/store/mod.rs [hunk 2]
```diff
@@ -60,12 +125,48 @@ impl Store {
...
+        let sync_seqn = sync.sync_seqn;
+
         self.shared.values.commit(tx.batch);
-        self.shared.pages.commit(tx.new_pages)?;
+        let prev_wal_size = self.shared.pages.sync_begin(tx.new_pages, sync_seqn)?;
         if let Some(new_root) = tx.new_root {
             *self.shared.root.lock() = new_root;
         }
-        self.shared.values.sync();
+        let data = self.shared.values.prepare_sync();
```
Comment (rphmeier) [inline nomt/src/store/mod.rs:139]: the inconsistency between `Beatree::prepare_sync` and `Bitbox::sync_begin`,  both in terminology and order, is slightly grating

Comment (pepyakin) [reply nomt/src/store/mod.rs:139]: my bad. Not even sure if either of those describes it.

**Pull request merged**